### PR TITLE
Document manual psql ALTER SYSTEM tuning instead of -c flags

### LINKED ISSUE #13: Apply server-side tuning programmatically as part of the import lifecycle
## Context

The tool's selling point is the one-shot `uvx --from ... musicbrainz-database-setup run ...` command. But the [Server-side tuning (optional)](https://github.com/rafacm/musicbrainz-database-setup/blob/main/docs/README.md#server-side-tuning-optional) section pulls the user out of that one-shot flow: they have to decide on tuning, recreate their Postgres container with `-c` flags, run the tool, and revert afterward.

The friction is sharpest when the user's Postgres lives in a committed `docker-compose.yml` they'd rather not edit (e.g. a shared instance hosting other databases). In that case the tuning is essentially "take it or leave it" — most users will leave it and accept a slow import.

## Proposal

Have the tool apply and revert the tuning itself, behind an opt-in flag (`--auto-tune`):

```bash
uvx --from git+https://github.com/rafacm/musicbrainz-database-setup \
  musicbrainz-database-setup run \
  --db postgresql://... \
  --modules core \
  --latest \
  --auto-tune
```

### Lifecycle

**Before import** — connect to the target DB and run:

```sql
ALTER SYSTEM SET shared_buffers = '2GB';
ALTER SYSTEM SET max_wal_size = '8GB';
ALTER SYSTEM SET checkpoint_timeout = '30min';
ALTER SYSTEM SET effective_io_concurrency = 200;
ALTER SYSTEM SET random_page_cost = 1.1;
SELECT pg_reload_conf();
```

**After import** (success **or** failure — wrap the whole import in try/finally) — revert:

```sql
ALTER SYSTEM RESET shared_buffers;
ALTER SYSTEM RESET max_wal_size;
ALTER SYSTEM RESET checkpoint_timeout;
ALTER SYSTEM RESET effective_io_concurrency;
ALTER SYSTEM RESET random_page_cost;
SELECT pg_reload_conf();
```

### The `shared_buffers` restart problem

`shared_buffers` is `postmaster`-only: it requires a full Postgres restart, not just `pg_reload_conf()`. The tool can't restart a Postgres it doesn't own — too many runtime variants (Docker container, `systemd`, Homebrew service, managed cloud DB).

Three reasonable handling options:

1. **Skip `shared_buffers` by default.** The other four flags are SIGHUP-reloadable and give the bulk of the speedup (the dominant gain is `max_wal_size` + `checkpoint_timeout` reducing checkpoint thrashing during COPY). Fully automatic, no manual intervention.
2. **Print and wait.** After ALTER SYSTEM, prompt: "Restart Postgres now (e.g. `docker compose restart postgres`), then press Enter to continue." Keeps `shared_buffers` in the set but breaks the unattended one-shot.
3. **Optional Docker integration.** If the user passes e.g. `--postgres-container <name>`, the tool can `docker restart <name>` itself. Covers the most common case (Docker Compose) without taking on `systemd` / Homebrew / cloud-managed restart logic.

**My preference:** (1) as the default behavior of `--auto-tune`, with (3) as an optional add-on (`--auto-tune --postgres-container ragtime-postgres-1` enables `shared_buffers` + automatic restart). (2) is a fallback if the user wants the full set but doesn't have Docker integration.

### Failure modes to handle

- **Tool crashes or is interrupted mid-import.** ALTER SYSTEM settings would persist into the user's normal config. Mitigations:
  - Wrap the import in try/finally (or an atexit handler) so RESET runs on every exit path.
  - Write a marker file (e.g. `~/.cache/musicbrainz-database-setup/tuning-applied`) at apply time; on next start, detect a stale marker and offer to revert before doing anything else.
- **User lacks ALTER SYSTEM privilege.** Requires superuser, or the `pg_alter_system` predefined role on PG16+. Detect upfront and degrade gracefully — log a warning, skip tuning, continue with the import.
- **Connection-user mismatch.** Tuning is server-wide, not per-database; the connection used for tuning doesn't have to be the import connection, but the tool should make sure the user understands that the settings affect every database on that instance for the duration.

## Why this fits the tool's design

The README already advertises a one-shot import. Tuning is currently the only documented step that breaks that promise. Making it programmatic and opt-in preserves the one-shot story for users who want it, while keeping the door open for those who prefer to manage tuning themselves.

### PR DESCRIPTION
**Pull request by rafacm**
## Summary

- Replace the existing `docker run … -c shared_buffers=2GB -c max_wal_size=8GB …` block in `docs/README.md` "Server-side tuning (optional)" with two `psql` `ALTER SYSTEM` blocks (apply, revert) bracketing the `run` command — four SIGHUP-reloadable knobs (`max_wal_size=8GB`, `checkpoint_timeout=30min`, `effective_io_concurrency=200`, `random_page_cost=1.1`), no Postgres restart required, so the optimization works against an existing instance the user doesn't want to recreate.
- `shared_buffers=2GB` drops out of the documented path (postmaster-only, restart-only) and is mentioned only as a 💡 aside for power users who control their Postgres startup.
- Rewrite the ⚠️ "Memory sizing" callout to anchor on `maintenance_work_mem` rather than `shared_buffers`; the OOM-kill anecdote from 2026-04-24 is preserved as a conditional warning. Update the README quick-start 💡 callout wording accordingly.
- Resolves #13 — a `--auto-tune` CLI flag was considered and rejected in favor of this docs-only path. The full design pass (privilege preflight, lifecycle bracketing, revert-failure handling, partial-apply atomicity, `shared_buffers`-restart edge cases) is preserved in [`docs/sessions/2026-04-30-manual-tuning-statements-planning-session.md`](docs/sessions/2026-04-30-manual-tuning-statements-planning-session.md).

## Test plan

Documentation-only change. CI is unaffected.

- [x] Markdown render check on `docs/README.md` — two fenced `psql` blocks bracketing the `run` command, preserved per-setting rationale, rewritten memory-sizing callout, no broken anchors.
- [x] `README.md` quick-start 💡 callout still resolves to `docs/README.md#server-side-tuning-optional`.
- [x] End-to-end smoke against a fresh `postgres:17-alpine`: `SHOW max_wal_size` is `1GB` → run apply block → `SHOW max_wal_size` is `8GB` → `musicbrainz-database-setup run …` completes → run revert block → `SHOW max_wal_size` is `1GB`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -35,25 +35,40 @@ The tool also works against managed PostgreSQL (RDS, Cloud SQL, etc.) as long as
...
+ALTER SYSTEM SET max_wal_size = '8GB';
+ALTER SYSTEM SET checkpoint_timeout = '30min';
+ALTER SYSTEM SET effective_io_concurrency = 200;
+ALTER SYSTEM SET random_page_cost = 1.1;
+SELECT pg_reload_conf();
+SQL
+
+# Run the import
+uvx --from git+https://github.com/rafacm/musicbrainz-database-setup \
+  musicbrainz-database-setup run --db "$DB_URL" --modules core --latest
```
Comment (rafacm) [inline docs/README.md:57]: We are using the `$DB_URL` environment variable here but we do not seem to define it anywhere (?).

Comment (rafacm) [reply docs/README.md:57]: Good catch. Added a single `export DB_URL=postgresql://postgres:postgres@localhost:5432/postgres` line above the first `psql` block so the snippet is self-contained — matches the literal connection string the README quick-start uses, and the three subsequent commands compose against it cleanly. Pushed in 010d8bd.
